src/nfp_devlink.c

```diff
@@ -121,8 +123,12 @@ nfp_devlink_port_split(struct devlink *devlink, unsigned int port_index,
 		goto out;
 	}
 
-	ret = nfp_devlink_set_lanes(pf, eth_port.index,
-				    eth_port.port_lanes / count);
+	lanes = eth_port.port_lanes / count;
```

Comment:
Would you mind doing:
```
lanes = eth_port.port_lanes / count;
/* Special case the 100G CXP -> 2x40G split */
if (eth_port.port_lanes == 10 && count == 2)
        lanes = 4;
```
i.e. be more explicit about what's going on?

Reply:
Note that I requested adding the count == 2 to the condition, in the future we may support breaking out to 10 10G ports, we'll want that not to fall under this work around.

Reply:
Sorry, got fixated on the comment. Updated and rebased.